lisa/environment.py

```diff
@@ -139,3 +146,40 @@ def _validate_single_default(
                 raise LisaException("only one node can set isDefault to True")
             has_default = True
         return has_default
+
+
+def load_environments(config: Dict[str, object]) -> None:
+    if not config:
+        raise LisaException("environment section must be set in config")
+    global max_concurrency
+    max_concurrency = cast(int, config.get(constants.ENVIRONMENT_MAX_CONCURRENDCY, 1))
+    environments_config = cast(
+        List[Dict[str, object]], config.get(constants.ENVIRONMENTS)
+    )
+    without_name: bool = False
+    log = get_logger("init", "env")
+    for environment_config in environments_config:
+        environment = Environment.load(environment_config)
+        if not environment.name:
+            if without_name:
+                raise LisaException("at least two environments has no name")
+            environment.name = _default_no_name
+            without_name = True
+        log.info(f"loaded environment {environment.name}")
+        _all_environments[environment.name] = environment
+
+
+def get_environments() -> Dict[str, Environment]:
+    return _all_environments
+
+
+def get_environment(name: Optional[str] = None) -> Environment:
```

Comment:
each test case can access its one environment only, won't be exposed to user. The name of env is used to force an environment in config. But I'm interested the pattern you described. Do you mean provide access to environments dict, so user can access by this way, or some metadata grammar can help on this?

Reply:
If you define `__getitem__` and `__setitem__` like [here](https://docs.python.org/3/reference/datamodel.html?emulating-container-types#emulating-container-types) you can make any class provide the `[foo]` interface for getting/sitting. It fits well for things that are like dicts (which this seems to be). It's best for classes that a user expects to act _like_ a dict, instead of making up a new interface with `get` and `add` methods (not that those aren't unreasonable, they're just not _as_ familiar as providing the built-in syntax the standard library uses).

Reply:
fixed